Runtime/Interaction/SnapZone.cs

```diff
@@ -218,6 +223,12 @@ private void OnDrawGizmos()
 
         protected virtual void Update()
         {
+            if (initialParent != null)
```

Comment:
Any particular reason to have it here and not in Start() or Update()?

Reply:
Yes.

The interactables register all their child colliders as part of it in the interaction manager at their Awake.

The idea is to deattach the snap zone from the interactable object and reattach it at the next first frame.

Unfortunately, the Awake and OnEnable from all interactors are called before than the Awake and OnEnable from all interactables so Awake and OnEnable are not an optional.

Additionally, Sockets already have a private Start, so calling a start will override their own and since it is private, there is no way to call as base.

The workaround I found is to call only once in the first frame in the Update, I don't entirely like it but, we can change it if the XR interaction framework becomes less private in the future.